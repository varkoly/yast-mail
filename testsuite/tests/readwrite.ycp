/**
 * Module:
 *   NIS client configuration
 *
 * Summary:
 *   Testsuite
 *
 * Authors:
 *   Martin Vidner <mvidner@suse.cz>
 *
 * $Id$
 */
{
    include "testsuite.ycp";
    import "Pkg"; // override
    import "Progress";
    import "Mail";

    map READ = $[
	// Runlevel:
	"init": $[
	    "scripts": $[
		"exists": true,
		"runlevel": $[
		    "sendmail": $[
			"start": [ "3", "5"],
			"stop":	 [ "3", "5"],
			],
		    "postfix": $[
			"start": [ "3", "5"],
			"stop":	 [ "3", "5"],
			],
		    "amavis": $[
			"start": [ "3", "5"],
			"stop":	 [ "3", "5"],
			],
		    ],
		// their contents is not important for ServiceAdjust
		"comment": $[
		    "sendmail": $[],
		    "postfix": $[],
		    "amavis": $[],
		    ],
		],
	    ],
	// Mail itself:
	"sysconfig": $[
	    "mail": $[
		"MAIL_CREATE_CONFIG": "yes",
		"SMTPD_LISTEN_REMOTE": "yes",
		"FROM_HEADER": "foo.test",
		],
	    "amavis": $[
		"USE_AMAVIS": "yes",
		],
	    "sendmail": $[
		"SENDMAIL_NOCANONIFY": "no",
		"SENDMAIL_EXPENSIVE": "no",
		"SENDMAIL_LOCALHOST": "",
		"SENDMAIL_SMARTHOST": "",
		"MASQUERADE_DOMAINS": "",
		// TODO: check for a new default in the installed system?
		// (evade the dummy agent, neededforbuild sendmail)
		"SENDMAIL_ARGS": "-L sendmail -Am -bd -q30m -om",
		"SMTP_AUTH_MECHANISMS": "",
		],
	    ],
	"mail": $[
	    "sendmail": $[
		"generics": $[
		    "table": [
			],
		    ],
		"virtuser": $[
		    "table": [
			],
		    ],
		"auth": $[
		    "accounts": [
			$["server": "s", "user": "u", "password": "p",
			  "more": 1],
			],
		    ],
		],
	    "aliases": $[
		"table": [
		    ],
		],
	    "fetchmail": $[
		"accounts": [
		    ],
		],
	    ],
	];

    map WRITE = $[
	];

    map EXECUTE = $[
	"target": $[
	    "bash_output": $[
		"exit": 0,
		"stdout": "",
		"stderr": "",
		],
	    ],
	];

    Pkg::FAKE (`IsProvided, $["sendmail": true, "amavis-sendmail": true,
			      "amavis-postfix": false]);

    block dont_abort = ``{return false;};
    Progress::off ();
    TEST (``(Mail::Read (dont_abort)), [READ, WRITE, EXECUTE], nil);
    Mail::write_only = false;
    TEST (``(Mail::Write (dont_abort)), [READ, WRITE, EXECUTE], nil);
}
