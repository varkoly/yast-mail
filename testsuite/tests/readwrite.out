Log	Pkg::`IsProvided (sendmail): true
Read	.sysconfig.mail.MAIL_CREATE_CONFIG "yes"
Read	.sysconfig.mail.SMTPD_LISTEN_REMOTE "yes"
Read	.sysconfig.sendmail.SENDMAIL_NOCANONIFY "no"
Read	.sysconfig.sendmail.SENDMAIL_EXPENSIVE "no"
Log	Pkg::`IsProvided (amavis-sendmail): true
Read	.sysconfig.amavis.USE_AMAVIS "yes"
Read	.sysconfig.sendmail.SENDMAIL_LOCALHOST ""
Read	.sysconfig.sendmail.SENDMAIL_SMARTHOST ""
Read	.sysconfig.mail.FROM_HEADER "foo.test"
Read	.sysconfig.sendmail.MASQUERADE_DOMAINS ""
Read	.mail.sendmail.generics.table []
Read	.mail.fetchmail.accounts []
Read	.mail.aliases.table []
Read	.mail.sendmail.virtuser.table []
Return	true
Write	.sysconfig.mail.MAIL_CREATE_CONFIG "yes" true
Write	.sysconfig.mail.SMTPD_LISTEN_REMOTE "yes" true
Write	.sysconfig.sendmail.SENDMAIL_NOCANONIFY "no" true
Write	.sysconfig.sendmail.SENDMAIL_EXPENSIVE "no" true
Read	.init.scripts.exists "sendmail" true
Read	.init.scripts.exists "sendmail" true
Read	.init.scripts.runlevel "sendmail" $["amavis":$["start":["3", "5"], "stop":["3", "5"]], "postfix":$["start":["3", "5"], "stop":["3", "5"]], "sendmail":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.comment "sendmail" $["amavis":$[], "postfix":$[], "sendmail":$[]]
Write	.sysconfig.amavis.USE_AMAVIS "yes" true
Read	.sysconfig.sendmail.SENDMAIL_ARGS "-L sendmail -Am -bd -q30m -om"
Write	.sysconfig.sendmail.SENDMAIL_LOCALHOST "" true
Write	.sysconfig.sendmail.SENDMAIL_SMARTHOST "" true
Write	.sysconfig.mail.FROM_HEADER "foo.test" true
Write	.sysconfig.sendmail.MASQUERADE_DOMAINS "" true
Write	.mail.sendmail.generics.table [] true
Write	.mail.aliases.table [] true
Write	.mail.sendmail.virtuser.table [] true
Write	.mail.fetchmail.accounts [] true
Write	.mail.fetchmail nil true
Write	.sysconfig.mail nil true
Write	.sysconfig.amavis nil true
Write	.sysconfig.sendmail nil true
Write	.mail.sendmail.generics nil true
Write	.mail.aliases nil true
Write	.mail.sendmail.virtuser nil true
Execute	.target.bash "/sbin/SuSEconfig --module sendmail" 0
Execute	.target.bash "/etc/init.d/amavis stop" $["TERM":"raw"] 0
Execute	.target.bash "/etc/init.d/amavis start" $["TERM":"raw"] 0
Read	.init.scripts.exists "amavis" true
Read	.init.scripts.exists "amavis" true
Read	.init.scripts.runlevel "amavis" $["amavis":$["start":["3", "5"], "stop":["3", "5"]], "postfix":$["start":["3", "5"], "stop":["3", "5"]], "sendmail":$["start":["3", "5"], "stop":["3", "5"]]]
Read	.init.scripts.comment "amavis" $["amavis":$[], "postfix":$[], "sendmail":$[]]
Execute	.target.bash "/etc/init.d/sendmail restart" $["TERM":"raw"] 0
Return	true
